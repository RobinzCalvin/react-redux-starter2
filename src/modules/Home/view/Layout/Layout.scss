.index-page {
  padding: 0 20px;

  &__description {
    font-size: 18px;
  }

  &__tool {
    margin-left: 10px;
  }

  &__content {
    margin: 0 auto;
    max-width: 60%;
  }

  &__github-icon {
    margin-right: 10px;
  }
}
