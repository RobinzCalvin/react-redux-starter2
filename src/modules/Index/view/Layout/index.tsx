import * as React from 'react';
import * as block from 'bem-cn';
import { Navbar, Form } from 'react-bootstrap';
import RowsLayout from '../../../../shared/view/elements/RowsLayout';
import Header from '../../../../shared/view/components/Header/index';
import { SearchRepositoriesInput } from '../../../../features/searchRepositories';
import Description from './Description';
import Search from './Search';
import * as s from './styles.styl';

interface Props {}

class IndexLayout extends React.Component<Props, null> {

    private b = block('index-page');

    constructor(props : Props) {
        super(props);
    }

    render () {
        const b = this.b;

        return (
            <RowsLayout
                footerContent={
                    <a href="http://fullstack-development.com/">FullStackDevelopment</a>
                }
                headerContent={
                    <Header>
                        <Navbar.Form pullRight>
                            <SearchRepositoriesInput />
                        </Navbar.Form>
                    </Header>
                }
            >
                <div className={s[b()]}>
                    <div className={s[b('content')]}>
                        <Description />
                        <Search />
                    </div>
                </div>
            </RowsLayout>
        );
    }
}

export {Props};
export default IndexLayout;