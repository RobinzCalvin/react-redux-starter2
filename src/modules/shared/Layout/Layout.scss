@import '~shared/styles/variables';

.layout {
  $header-footer-height: 3rem;
  $header-footer-margin: 1.5rem;

  position: relative;
  padding-bottom: $header-footer-height + $header-footer-margin;
  min-height: 100%;

  &__title {
    text-align: center;
    font-weight: 500;

    @media (max-width: $change-view-breakpoint) {
      font-size: 1.7rem;
    }
  }

  &__header {
    margin-bottom: $header-footer-margin;
    height: $header-footer-height;
    background: #333;
    color: #eaeaea;

    @media (max-width: $change-view-breakpoint) {
      justify-content: center;
    }

    &-content {
      display: flex;
      align-items: center;
      justify-content: space-between;
      margin: auto;
      width: 90%;
    }

    &-title {
      @media (max-width: $change-view-breakpoint) {
        display: none;
      }
    }

    &-left-part {
      display: flex;
      align-items: center;
    }
  }

  &__tabs {
    display: flex;
    margin-left: 1rem;
    height: 100%;
    line-height: $header-footer-height;

    @media (max-width: $change-view-breakpoint) {
      margin-left: 0;
    }
  }

  &__tab {
    width: 8rem;
    background: #333;
    text-align: center;

    &_active {
      background: #555;
    }
  }

  &__navigation-link {
    color: #eaeaea;
    text-decoration: none;
  }

  &__content {
    display: flex;
    flex-direction: column;
    justify-content: center;
    min-height: 77vh;
  }

  &__footer {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: $header-footer-height;
    background: #333;
    text-align: right;
    line-height: $header-footer-height;

    &-content {
      margin: auto;
      width: 90%;
      text-align: end;

      @media (max-width: $change-view-breakpoint) {
        width: 100%;
        text-align: center;
      }
    }
  }

  &__company-link {
    color: #eaeaea;
  }
}
