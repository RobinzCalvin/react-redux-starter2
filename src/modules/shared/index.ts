export { default as Layout } from './Layout/Layout';
export { default as withLayout } from './withLayout';
