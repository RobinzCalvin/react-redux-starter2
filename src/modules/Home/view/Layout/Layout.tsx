import * as React from 'react';
import * as block from 'bem-cn';
import { RouteComponentProps } from 'react-router-dom';
import { Navbar } from 'react-bootstrap';
import RowsLayout from '../../../../shared/view/elements/RowsLayout';
import Header from '../../../../shared/view/components/Header/index';
import * as loadSearchRepositories from '../../../../features/searchRepositories/entry';
import Description from './Description';
import Search from './Search';
import { featureConnect } from 'core';
import './Layout.scss';
import { BundleLoader } from 'shared/types/app';

interface IOwnProps {
  searchRepositoriesEntry: loadSearchRepositories.Entry;
}

type Props = IOwnProps & RouteComponentProps<void>;

class HomeLayout extends React.PureComponent<Props, {}> {
  private b = block('index-page');

  public render() {
    const b = this.b;
    const { history } = this.props;
    const { SearchRepositoriesInput } = this.props.searchRepositoriesEntry.containers;

    return (
      <RowsLayout
        footerContent={<a href="http://fullstack-development.com/">FullStackDevelopment</a>}
        headerContent={(
          <Header onLinkClick={history.push}>
            <Navbar.Form pullRight>
              <SearchRepositoriesInput />
            </Navbar.Form>
          </Header>
        )}
      >
        <div className={b()}>
          <div className={b('content')()}>
            <Description />
            12234556
            <Search SearchRepositoriesInput={SearchRepositoriesInput} />
          </div>
        </div>
      </RowsLayout>
    );
  }
}

const withFeatures = featureConnect({
  searchRepositoriesEntry: loadSearchRepositories as any as BundleLoader<any>,
})(HomeLayout);

export default withFeatures;
