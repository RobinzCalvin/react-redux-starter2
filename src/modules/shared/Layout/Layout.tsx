import * as React from 'react';
import block from 'bem-cn';
import { Link, withRouter, RouteComponentProps } from 'react-router-dom';

import { featureConnect } from 'core';
import * as features from 'features';
import routes from '../../routes';
import './Layout.scss';

interface IOwnProps {
  title: string;
}

interface IFeatureProps {
  profileFeatureEntry: features.profile.Entry;
}

type IProps = IOwnProps & IFeatureProps & RouteComponentProps & React.ComponentProps<'div'>;

const b = block('layout');

function Layout(props: IProps) {
  const { children, location, title, profileFeatureEntry: { containers } } = props;
  const { ProfilePreview } = containers;

  return (
    <div className={b()}>
      <header className={b('header')}>
        <div className={b('header-left-part')}>
          <span className={b('header-title')}>Search for:</span>
          <div className={b('tabs')}>
            {renderTab(routes.search.users.getRoutePath(), 'Users')}
            {renderTab(routes.search.repositories.getRoutePath(), 'Repositories')}
          </div>
        </div>
        <ProfilePreview onEditClick={handleEditProfileClick} />
      </header>
      <div className={b('content')}>
        <h1 className={b('title')}>
          {title}
        </h1>
        {children}
      </div>
      <footer className={b('footer')}>
        <a className={b('company-link')} href="https://fullstack-development.com" target="_blank">
          Fullstack Development
        </a>
      </footer>
    </div>
  );

  function renderTab(path: string, tabTitle: string) {
    return (
      <Link to={path} className={b('navigation-link')}>
        <div className={b('tab', { active: path === location.pathname })}>
          {tabTitle}
        </div>
      </Link>
    );
  }

  function handleEditProfileClick() {
    const { history } = props;
    history.push(routes.profile.getRoutePath());
  }
}

export default featureConnect({
  profileFeatureEntry: features.profile.loadEntry,
})(withRouter(Layout));
