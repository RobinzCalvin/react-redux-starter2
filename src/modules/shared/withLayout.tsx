import React from 'react';

import Layout from './Layout/Layout';

function withLayout(title: string) {
  return (Component: React.ComponentType) => () => (
    <Layout title={title}>
      <Component />
    </Layout>
  );
}

export default withLayout;
