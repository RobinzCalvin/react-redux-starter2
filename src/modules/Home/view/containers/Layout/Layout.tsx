import * as React from 'react';
import { featureConnect } from 'core';
import { Navbar } from 'react-bootstrap';
import { RouteComponentProps } from 'react-router-dom';

import * as searchRepositories from 'features/searchRepositories';

import { RowsLayout } from 'shared/view/elements';
import { Header, Footer } from 'shared/view/components';

import { homeRedirectPath, orderRedirectPath } from '../../../../routes';
import { Description, Search } from '../../components';
import { StylesProps, provideStyles } from './Layout.style';

interface IOwnProps {
  searchRepositoriesEntry: searchRepositories.Entry;
}

type Props = IOwnProps & StylesProps & RouteComponentProps<void>;

class HomeLayout extends React.PureComponent<Props> {
  public render() {
    const { classes, searchRepositoriesEntry } = this.props;
    const { SearchRepositoriesInput } = searchRepositoriesEntry.containers;

    return (
      <RowsLayout
        footerContent={<Footer />}
        headerContent={(
          <Header
            brandRedirectPath={homeRedirectPath}
            menuRedirectPaths={{ order: orderRedirectPath }}
          >
            <Navbar.Form pullRight>
              <SearchRepositoriesInput />
            </Navbar.Form>
          </Header>
        )}
      >
        <div className={classes.content}>
          <div className={classes.description}><Description /></div>
          <Search SearchRepositoriesInput={SearchRepositoriesInput} />
        </div>
      </RowsLayout>
    );
  }
}

export default (
  featureConnect({ searchRepositoriesEntry: searchRepositories.loadEntry })(
    provideStyles(HomeLayout),
  )
);
