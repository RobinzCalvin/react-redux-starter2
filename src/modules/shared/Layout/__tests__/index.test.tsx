import React from 'react';
import { shallow } from 'enzyme';

import { makeMockEntry, withRouterProps } from 'shared/mocks';

import { Layout, ILayoutProps } from '../Layout';

const props: ILayoutProps = {
  title: 'Title',
  profileFeatureEntry: makeMockEntry(),
  ...withRouterProps,
};

describe('Layout component', () => {
  it('should match snapshot', () => {
    const component = shallow(<Layout {...props} />);
    expect(component).toMatchSnapshot();
  });
});
