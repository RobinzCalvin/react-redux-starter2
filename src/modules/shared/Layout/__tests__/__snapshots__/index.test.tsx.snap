// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should match snapshot 1`] = `
<div
  className="layout"
>
  <header
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <LayoutHeaderMenu
          menuItems={
            Array [
              Object {
                "path": "/search/users",
                "title": "Users",
              },
              Object {
                "path": "/search/repositories",
                "title": "Repositories",
              },
            ]
          }
        />
      </div>
      <Component
        onEditClick={[Function]}
      />
    </div>
  </header>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <h1
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      Title
    </h1>
  </div>
  <footer
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <a
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
        href="https://fullstack-development.com"
        target="_blank"
      >
        Fullstack Development
      </a>
    </div>
  </footer>
</div>
`;
